src/engine/enginebuffer.cpp

```diff
@@ -1134,11 +1150,13 @@ void EngineBuffer::process(CSAMPLE* pOutput, const int iBufferSize) {
 void EngineBuffer::processSlip(int iBufferSize) {
     // Do a single read from m_bSlipEnabled so we don't run in to race conditions.
     bool enabled = static_cast<bool>(load_atomic(m_slipEnabled));
+    bool cancelled = static_cast<bool>(load_atomic(m_slipCancelled));
     if (enabled != m_bSlipEnabledProcessing) {
         m_bSlipEnabledProcessing = enabled;
-        if (enabled) {
+        if (enabled || cancelled) {
             m_dSlipPosition = m_filepos_play;
             m_dSlipRate = m_rate_old;
+            m_slipCancelled = 0;
```

Comment:
I do not know if this race condition can happen, but at least there is a chance. 
This can be avoided by a second Atomic integer for the "Processed" state or just using fetchAndStoreAcquire() to reset and read m_slipCancelled at once.